tests/crypto/cpu_entropy_test.c

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#define MAX_LOOP_SIZE 1000
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/rdrand.h>
+#include <openenclave/internal/rdseed.h>
+#include <openenclave/internal/tests.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+// Test that RDSEED and RDRAND functions synchronously block and
+// retry until the sufficient entropy exists to be returned.
+void TestCpuEntropy()
+{
+    uint64_t rand_num = 0;
+    printf("=== begin %s()\n", __FUNCTION__);
+
+    /* TODO: This test does not actually manage to exhaust the RDRAND
+     * entropy pool regardless of the number of iterations run since
+     * the operation of retrieving the RDRAND value through the bus
+     * architecture is slow enough that a single thread can't saturate
+     * the interface regardless of the number of iterations run.
+     */
+    for (uint64_t i = 0; i < MAX_LOOP_SIZE; i++)
+    {
+        rand_num = oe_rdrand();
+
+        /* 0 is a legal random value that could be returned, but the
+         * odds of this happening twice in a row are very unlikely
+         * unless we've run out of hardware entropy and are returning
+         * without retrying until we have sufficient entropy.
+         */
+        if (rand_num == 0)
```

Comment:
Anita added this test and I tried to reverse engineer the logic behind it, hence the added comment. My understanding is:
* The test is trying to saturate the underlying interface and force it to loop before returning.
* `rand_num == 0` does not necessarily indicate a failure of the `oe_rdrand` loop logic since success/failure is indicated by the CF 1/0, not the value of `rand_num` itself, so the test can't just assume `oe_rand` is failing to loop if it sees a random value of 0.
* If it does see two consecutive zero random values, then it assumes that the odds of that happening are so small, that it does suggest that `oe_rdrand` failed to block and retry. It's hokey, but I'm assuming it was an issue Anita encountered when this was running a 100 mil iterations at a time, and that this change made it more robust.
* I'm just keeping this as an automated sanity check that can be walked using a debugger at this point, see `TODO` regarding reducing the number of iterations it now runs.